File: CS/MultiGroupSummary/ToolTipHelper.cs
using DevExpress.Data;
using DevExpress.Utils;
using DevExpress.XtraGrid;
using Q354185.MyXtraGrid;
using System;
using System.Collections.Generic;
using System.Linq;

namespace MultiGroupSummary {
    public class ToolTipHelper {
        private MyGridControl gridCtrl;
        public ToolTipHelper(MyGridControl myGridControl1) {
            Init(myGridControl1);
        }
        private void Init(MyGridControl gridControl) {
            gridCtrl = gridControl;
            var toolTipController = new ToolTipController();
            gridCtrl.ToolTipController = toolTipController;
            toolTipController.GetActiveObjectInfo += new ToolTipControllerGetActiveObjectInfoEventHandler(ToolTipController_GetActiveObjectInfo);
        }
        private void ToolTipController_GetActiveObjectInfo(object sender, ToolTipControllerGetActiveObjectInfoEventArgs e) {
            ToolTipControlInfo info = null;
            var view = gridCtrl.GetViewAt(e.ControlMousePosition) as MyGridView;
            if (view == null) {
                return;
            }
            var hitInfo = view.CalcHitInfo(e.ControlMousePosition);
            if (hitInfo == null) {
                return;
            }
            var hitInfoFooterCell = hitInfo.FooterCell;
            if (hitInfoFooterCell != null) {
                var summaryItem = new GridGroupSummaryItem();
                var tag = hitInfoFooterCell.ColumnInfo.Tag as GridGroupSummaryItem;
                if (tag != null) {
                    summaryItem = tag;
                } else {
                    var rowSummaryItem = view.GetRowSummaryItem(hitInfo.RowHandle, hitInfoFooterCell.Column);
                    summaryItem = rowSummaryItem.Key as GridGroupSummaryItem;
                }
                if (summaryItem == null) {
                    return;
                }
                if (summaryItem.SummaryType == SummaryItemType.None) {
                    return;
                }
                info = new ToolTipControlInfo(hitInfoFooterCell.Value, hitInfoFooterCell.DisplayText);
            }
            if (info == null) {
                return;
            }
            e.Info = info;
        }
    }
}

File: VB/MultiGroupSummary/ToolTipHelper.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Data
Imports DevExpress.Utils
Imports DevExpress.XtraGrid
Imports Q354185.MyXtraGrid
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace MultiGroupSummary
	Public Class ToolTipHelper
		Private gridCtrl As MyGridControl
		Public Sub New(ByVal myGridControl1 As MyGridControl)
			Init(myGridControl1)
		End Sub
		Private Sub Init(ByVal gridControl As MyGridControl)
			gridCtrl = gridControl
			Dim toolTipController = New ToolTipController()
			gridCtrl.ToolTipController = toolTipController
			AddHandler toolTipController.GetActiveObjectInfo, AddressOf ToolTipController_GetActiveObjectInfo
		End Sub
		Private Sub ToolTipController_GetActiveObjectInfo(ByVal sender As Object, ByVal e As ToolTipControllerGetActiveObjectInfoEventArgs)
			Dim info As ToolTipControlInfo = Nothing
			Dim view = TryCast(gridCtrl.GetViewAt(e.ControlMousePosition), MyGridView)
			If view Is Nothing Then
				Return
			End If
			Dim hitInfo = view.CalcHitInfo(e.ControlMousePosition)
			If hitInfo Is Nothing Then
				Return
			End If
			Dim hitInfoFooterCell = hitInfo.FooterCell
			If hitInfoFooterCell IsNot Nothing Then
				Dim summaryItem = New GridGroupSummaryItem()
				Dim tag = TryCast(hitInfoFooterCell.ColumnInfo.Tag, GridGroupSummaryItem)
				If tag IsNot Nothing Then
					summaryItem = tag
				Else
					Dim rowSummaryItem = view.GetRowSummaryItem(hitInfo.RowHandle, hitInfoFooterCell.Column)
					summaryItem = TryCast(rowSummaryItem.Key, GridGroupSummaryItem)
				End If
				If summaryItem Is Nothing Then
					Return
				End If
				If summaryItem.SummaryType = SummaryItemType.None Then
					Return
				End If
				info = New ToolTipControlInfo(hitInfoFooterCell.Value, hitInfoFooterCell.DisplayText)
			End If
			If info Is Nothing Then
				Return
			End If
			e.Info = info
		End Sub
	End Class
End Namespace
